internal/service/kendra/experience.go

```diff
@@ -0,0 +1,575 @@
+package kendra
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/service/kendra"
+	"github.com/aws/aws-sdk-go-v2/service/kendra/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceExperience() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceExperienceCreate,
+		ReadWithoutTimeout:   resourceExperienceRead,
+		UpdateWithoutTimeout: resourceExperienceUpdate,
+		DeleteWithoutTimeout: resourceExperienceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"content_source_configuration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							AtLeastOneOf: []string{
+								"configuration.0.content_source_configuration",
+								"configuration.0.user_identity_configuration",
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data_source_ids": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										MinItems: 1,
+										MaxItems: 100,
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringMatch(regexp.MustCompile(`[a-zA-Z0-9][a-zA-Z0-9_\-]*`), ""),
+										},
+									},
+									"direct_put_content": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
```

Comment:
important note: without the `data_source` and `faq` kendra resources in the provider at the time of writing, I was not able to verify if this default value may interfere in some way with one or both of the other args (`data_source_ids` and `faq_ids`); will need to verify with a test when the other resources are added